activerecord/test/cases/schema_dumper_test.rb

```diff
@@ -524,3 +501,68 @@ def test_schema_dump_with_float_column_infinity_default
     assert_match %r{t\.float\s+"float_with_nan_default",\s+default: ::Float::NAN}, output
   end
 end
+
+if current_adapter?(:SQLite3Adapter) && !in_memory_db?
```

Comment:
To be honest I don't really understand this code, but I copied it from https://github.com/rails/rails/blob/master/activerecord/test/cases/tasks/database_tasks_test.rb#L885..L905 and it does what I need.

What I want to do here is have `ActiveRecord::Base.connection` return a connection which has the `migrations_path` config I provide on line 515. I need this because my tests need to compare what's in the filesystem (at `MIGRATIONS_ROOT + "/schema_dumper"`) and what's in the db (done per-test). I'm open to better/simpler ways of doing this, particularly as my current approach only runs for SQLite tests.